specs/core/1_beacon-chain-misc.md

```diff
@@ -0,0 +1,218 @@
+    first_slot_in_last_crosslink = state.current_crosslinks[receipt_proof.shard].start_epoch * SLOTS_PER_EPOCH
+    gindex = concat_generalized_indices(
+        get_generalized_index_of_crosslink_header(0),
+        get_generalized_index(ShardBlockHeader, 'state_root'),
+        compute_historical_state_generalized_index(receipt_slot, first_slot_in_last_crosslink),
+        get_generalized_index(ShardState, 'receipt_root')
+    )
+    assert verify_merkle_proof(
+        leaf=hash_tree_root(receipt_proof.receipt),
+        proof=receipt_proof.proof,
+        index=gindex,
+        root=state.current_crosslinks[shard].data_root
+    )
+    for delta in receipt_proof.receipt:
+        if get_current_epoch(state) < state.validators[delta.index].withdrawable_epoch:
+            increase_balance(state, delta.index, state.validators[delta.index].effective_balance * delta.reward_coefficient // REWARD_COEFFICIENT_BASE)
+            decrease_balance(state, delta.index, delta.block_fee)
+    state.next_shard_receipt_period[receipt_proof.shard] += 1
+    increase_balance(state, get_beacon_proposer_index(state), MICRO_REWARD)
+```
+
+## Changes
+
+### Persistent committees
+
+Add to the beacon state the following fields:
+
+```python
+# begin insert @persistent_committee_fields
+    previous_persistent_committee_root: Hash
+    current_persistent_committee_root: Hash
+    next_persistent_committee_root: Hash
+    next_shard_receipt_period: Vector[uint, SHARD_COUNT]
+# end insert @persistent_committee_fields
+```
+`next_shard_receipt_period` values initialized to `PHASE_1_FORK_SLOT // SLOTS_PER_EPOCH // EPOCHS_PER_SHARD_PERIOD`
+
+Run `update_persistent_committee` immediately before `process_final_updates`:
+
+```python
+# begin insert @update_persistent_committee
+    update_persistent_committee(state)
+# end insert @update_persistent_committee
+def update_persistent_committee(state: BeaconState) -> None:
+    """
+    Updates persistent committee roots at boundary blocks.
+    """
+    if (get_current_epoch(state) + 1) % EPOCHS_PER_SHARD_PERIOD == 0:
+        state.previous_persistent_committee_root = state.current_persistent_committee_root
+        state.current_persistent_committee_root = state.next_persistent_committee_root
+        committees = Vector[CompactCommittee, SHARD_COUNT]([
+            committee_to_compact_committee(state, get_period_committee(state, get_current_epoch(state) + 1, i))
+            for i in range(SHARD_COUNT)
+        ])
+        state.next_persistent_committee_root = hash_tree_root(committees)
+```
+
+### Shard receipt processing
+
+Add the `shard_receipts` operation to `BeaconBlockBody`:
+
+```python
+# begin insert @shard_receipts
+    shard_receipts: List[ShardReceipt, MAX_SHARD_RECEIPTS]
+# end insert @shard_receipts
+```
+
+Use `process_shard_receipt` to process each receipt.
+
+```python
+# begin insert @process_shard_receipts
+        (body.shard_receipts, process_shard_receipts),
```

Comment:
`process_shard_receipt` takes a `ShardReceiptProof`. Should we instead define the operation added to the `BeaconState` as `shard_receipt_proofs`?